cmd/bucket-targets.go

```diff
@@ -428,3 +442,37 @@ func parseBucketTargetConfig(bucket string, cdata, cmetadata []byte) (*madmin.Bu
 	}
 	return &t, nil
 }
+
+// TargetClient is the struct for remote target client.
+type TargetClient struct {
+	*miniogo.Client
+	up                  int32
+	healthCheckDuration time.Duration
+	bucket              string // remote bucket target
+	replicateSync       bool
+}
+
+func (tc *TargetClient) setOffline() {
+	atomic.StoreInt32(&tc.up, 0)
+}
+
+func (tc *TargetClient) setOnline() {
+	atomic.StoreInt32(&tc.up, 1)
+}
+
+func (tc *TargetClient) isOffline() bool {
+	return atomic.LoadInt32(&tc.up) == 0
+}
+
```

Comment:
Take a look at how to do offline,online state as per `cmd/rest/client.go`

Reply:
updated